numpy/__init__.pyi

```diff
@@ -408,7 +408,15 @@ dstack: Any
 ediff1d: Any
 expand_dims: Any
 extract: Any
-eye: Any
+def eye(
+    N: int,
+    M: int = ...,
+    k: int = ...,
+    dtype: DTypeLike = ...,
+    order: str = ...,
+    *,
+    like: ArrayLike = ...
+) -> ndarray: ...
```

Comment:
I've got a couple of comments but the PR should be good to go afterwards.
```suggestion
def eye(
    N: int,
    M: Optional[int] = ...,
    k: int = ...,
    dtype: DTypeLike = ...,
    order: _OrderCF = ...,
    *,
    like: Optional[ArrayLike] = ...
) -> ndarray[Any, Any]: ...
```
* Both the `M` and `like` parameters can also be `None`.
* It's generally considered best practice to set subscriptable values to `Any` rather than leaving them unparameterized. 
* Since `order` can only take a few specific strings we can use string-literals here. https://github.com/numpy/numpy/blob/eb4317d0a4c94b3f1454dca58c76caed8bbdc7e6/numpy/__init__.pyi#L850

Reply:
Thanks for taking the time to fix my mistakes!

Reply:
There should be this `Commit suggestion` button here on github;
pressing it should apply automatically apply the suggested changes.